### PR TITLE
Introduce inject integration tests

### PR DESCRIPTION
**Pull request by siggy**
This change introduces integration tests for `linkerd inject`. The tests
perform CLI injection, with and without params, and validates the
output, including annotations.

Also add some known errors in logs to `install_test.go`.

TODO:
- deploy uninjected and injected resources to a default and
  auto-injected cluster
- test creation and update

Part of #2459

Signed-off-by: Andrew Seigner <siggy@buoyant.io>

### CODE REVIEW

**Comment by alpeb** [comment]
Actually I'm not sure about my previous comment anymore. All the integration tests rely on the setup done by `install_test.go`. So in this case a simple comment might suffice, clarifying that the control plane is configured for auto-injection at this point.

File: test/inject/inject_test.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+			if len(pods) != 1 {
+				t.Fatalf("expected 1 pod for namespace %s, got %d", ns, len(pods))
+			}
+
+			shouldBeInjected := false
+			switch nsAnnotation {
+			case "", k8s.ProxyInjectDisabled:
+				switch podAnnotation {
+				case k8s.ProxyInjectEnabled:
+					shouldBeInjected = true
```
[the remark this answers is not in the document]
Comment (siggy) [reply test/inject/inject_test.go]: It's completely subjective. I tried writing this both ways, and I find the `switch` slightly more readable than the `if`.

Here they are side-by-side, let me know what you think:

```go
shouldBeInjected := false
if nsAnnotation == "" || nsAnnotation == k8s.ProxyInjectDisabled {
	shouldBeInjected = podAnnotation == k8s.ProxyInjectEnabled
} else if nsAnnotation == k8s.ProxyInjectEnabled {
	shouldBeInjected = (podAnnotation == "" || podAnnotation == k8s.ProxyInjectEnabled)
}
```

```go
shouldBeInjected := false
switch nsAnnotation {
case "", k8s.ProxyInjectDisabled:
	switch podAnnotation {
	case k8s.ProxyInjectEnabled:
		shouldBeInjected = true
	}
case k8s.ProxyInjectEnabled:
	switch podAnnotation {
	case "", k8s.ProxyInjectEnabled:
		shouldBeInjected = true
	}
}
```

**Comment by siggy** [comment]
@alpeb Yup there's an assumption with the integration tests that `install_test.go` always runs first: https://github.com/linkerd/linkerd2/blob/38f504beb14cf216443ec7a6bad147d25c0d308e/TEST.md#running-tests

Agree that `inject_test.go` is still a little special, in that is specifically requires a control-plane with auto-injection. I've added a comment above `TestAnnotationPermutations` calling this out.

File: test/inject/inject_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alpeb) [reply test/inject/inject_test.go]: The outer switch looks good. I was gonna say `if`s are better in the inner conditions, but not so much for the second one where an _or_ is involved. But it's definitely more readable keeping all switches for consistency-sake. I'm gonna steal this idiom :shipit:

**Pull request merged**